spec/lib/traffic_source_parser/parser/referrer_parser_spec.rb

```diff
@@ -0,0 +1,217 @@
+          end
+        end
+    end
+  end
+
+  describe ".parse" do
+
+    context "when cookie value is (none)" do
+
+      let(:sources) do
+        [
+          {
+            cookie_value: "(none)",
+            source: "(direct)",
+            campaign: "(direct)",
+            channel: "Direct",
+            medium: "(none)"
+          }
+        ]
+      end
+
+      let(:klass) {  TrafficSourceParser::Result::Direct }
+
+      it_behaves_like "returns the correct object with attributes"
+
+    end
+
+    context "when cookie value is empty" do
+
+      let(:sources) do
+        [
+          {
+            cookie_value: '',
+            source: "unknown",
+            campaign: "unknown",
+            channel: "Unknown",
+            medium: "unknown"
+          }
+        ]
+      end
+
+      let(:klass) {  TrafficSourceParser::Result::Unknown }
+
+      it_behaves_like "returns the correct object with attributes"
+
+    end
+
+    context "when cookie value is nil" do
+
+      let(:sources) do
+        [
+          {
+            cookie_value: nil,
+            source: "unknown",
+            campaign: "unknown",
+            channel: "Unknown",
+            medium: "unknown"
+          }
+        ]
+      end
+
+      let(:klass) {  TrafficSourceParser::Result::Unknown }
+
+      it_behaves_like "returns the correct object with attributes"
+
+    end
+
+    context "when source is referrer" do
+
+      context "and isn't a recognized referrer" do
+
+          # source: domain (ajuda.rdstation.com.br)
+           # campaign: (referral)
+           # channel: "Referral",
+           # medium: referral
+           # content: /xunda
+
+
+           let(:sources) do
+            [
+              {
+                cookie_value: "http://shipit.resultadosdigitais.com.br/trabalhe-conosco/",
+                source:  "resultadosdigitais.com.br",
+                medium: "referral",
+                campaign: "(referral)",
+                channel: "Referral",
+                term: "/trabalhe-conosco/"
+                },
+                {
+                  cookie_value: "https://thoughtbot.com/locations",
+                  source:  "thoughtbot.com",
+                  medium: "referral",
+                  campaign: "(referral)",
+                  channel: "Referral",
+                  term: '/locations'
+
+                  },
+                  {
+                    cookie_value: "https://rubygems.org/",
+                    source:  "rubygems.org",
+                    medium: "referral",
+                    campaign: "(referral)",
+                    channel: "Referral",
+
+                    },
+                    {
+                      cookie_value: "https://www.ruby-lang.org/en/about/",
+                      source:  "ruby-lang.org",
+                      medium: "referral",
+                      campaign: "(referral)",
+                      channel: "Referral",
+                      term: '/en/about/'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.